### PR TITLE
NL-65 Notes Panel Backend

### LINKED ISSUE #65: Connect Notes Panel to Backend
## Name and Email
  - Name: Rickesh Khilnani
  - Email: rickeshkhilnani@gmail.com

## Complexity of Issue (Scale from 1-5)
  - 3
  
## Issue Description
  - Going to edit WeeklyJS.js

### PR DESCRIPTION
**Pull request by Rick10101221**
MUST MERGE [PR 68](https://github.com/CSE112-sp22-group9/cse112-sp22-group9/pull/68)
MUST MERGE [PR 56](https://github.com/CSE112-sp22-group9/cse112-sp22-group9/pull/56)
MUST MERGE [PR 54](https://github.com/CSE112-sp22-group9/cse112-sp22-group9/pull/54)

Closes #65 

Currently mergeable.

### CODE REVIEW

**Comment by Rick10101221** [comment]
Need to figure out when to update database for notes (when user navigates away from notes panel or on page change)

**Comment by cfellowes** [comment]
seems good, we just need to implement note saving next

File: source/Backend/BackendInit.js [hunk 1]
```diff
@@ -199,6 +336,7 @@ function getMonthlyGoals(monthStr) {
  * user is not signed in (i.e bypassing the authentication).
  */
 function getUserID() {
+    // source: https://github.com/firebase/firebase-js-sdk/issues/462#:~:text=you%20can%20easily%20implement%20that%20on%20your%20own%20with%20a%20couple%20of%20lines%3A
```
Comment (DustinLin) [inline source/Backend/BackendInit.js:340]: What is this link for?

**Comment by Yenyun035** [comment]
> Need to figure out when to update database for notes (when user navigates away from notes panel or on page change)

The indexedDB version updates notes whenever the "focusout" event of the notes element is triggered. Focusout event probably occurs when user clicks other areas in the page to leave the notes section.

File: source/Backend/BackendInit.js (hunk 1 above)
Comment (Rick10101221) [reply source/Backend/BackendInit.js:340]: It's there as an explanation of where I got the getUserID() code from. I can remove it if you'd like, but I put it there so other devs would not be confused

**Comment by Rick10101221** [comment]
Look into "focusout" event for notes. Currently unmergeable.

**Review by DustinLin — APPROVED** [review]

**Comment by Rick10101221** [comment]
Mergeable. Will include "focusout" or page-change-call to updateNotes() in another Issue/PR.

**Comment by DustinLin** [comment]
resolved merge conflicts

**Pull request merged**